keys.go

```diff
@@ -259,50 +259,36 @@ func (c *Client) DeleteKey(ctx context.Context, id string, prefer PreferReturn,
 	return nil, nil
 }
 
-// RestoreKey restores a deleted imported root key by specifying the ID of the key
-func (c *Client) RestoreKey(ctx context.Context, id, payload, encryptedNonce, iv string) (*Key, error) {
-
-	if payload == "" {
-		return nil, fmt.Errorf("Please provide payload to restore the key")
-	}
-
-	key := Key{
-		Payload:        payload,
-		IV:             iv,
-		EncryptedNonce: encryptedNonce,
-	}
-
-	if encryptedNonce != "" && iv != "" {
-		key.EncryptionAlgorithm = importTokenEncAlgo
-	}
-
-	keysRequest := Keys{
-		Metadata: KeysMetadata{
-			CollectionType: keyType,
-			NumberOfKeys:   1,
-		},
-		Keys: []Key{key},
-	}
-
-	v := url.Values{}
-	v.Set("action", "restore")
-
-	req, err := c.newRequest("POST", fmt.Sprintf("keys/%s", id), &keysRequest)
+// RestoreKeyWithoutPayload method reverts a delete key status to active key
+// This method performs restore of any key from deleted state to active state
+// without any additional parameters.
+// For more information please refer to the link below:
+// https://cloud.ibm.com/dowcs/key-protect?topic=key-protect-restore-keys
+func (c *Client) RestoreKeyWithoutPayload(ctx context.Context, id string) (*Key, error) {
```

Comment:
```suggestion
func (c *Client) RestoreKey(ctx context.Context, id string) (*Key, error) {
```

Lets just drop the old signature and use the new one, since the old path is totally dead and will never work anymore.

Reply:
@pbavari so Matt idea is since this is not a frequently used method. The customers can change their method call when  they error out. If we allow them to pass in payload , it will be a mislead as they will be expecting that payload to be set for their key. We will be having a release tag before merging this PR so that incase users are not ready to make the change, they can still use the older version of sdk.

Reply:
We need to document that as a breaking change for the release where this change goes in.

Reply:
yeah  the current one is 0.5.4 , so the next one will be the next release tag will be 0.6.0 with appropriate description.

Reply:
I planned to put that in the release notes yes.